docs/rfc/rfc-8.adoc

```diff
@@ -0,0 +1,482 @@
+  selects one of the L2 chains they want to bridge from.
+* The holder submits one transaction to L2. After the transaction is mined, they
+  wait about 15 minutes for the L2 block finality.
+* The holder submits one transaction to L1. After that transaction is mined,
+  they have their tBTC on L1.
+
+What is not immediately obvious is that the token received on L2 may not be the
+canonical token on that L2.
+
+For example, when I crossed the bridge with ETH, I received WETH token at
+address link:https://arbiscan.io/token/0xd8369c2eda18dd6518eabb1f85bd60606deb39ec[`0xd8369c2eda18dd6518eabb1f85bd60606deb39ec`].
+This is not the canonical WETH on Arbitrum. The canonical WETH on Arbitrum is
+link:https://arbiscan.io/token/0x82af49447d8a07e3bd95bd0d56f35241523fbab1[`0x82af49447d8a07e3bd95bd0d56f35241523fbab1`].
+When I crossed the bridge with USDC I received USDC token at
+link:https://arbiscan.io/token/0xc96f2715e2a242d50d1b0bc923dbe1740b8ecf18[`0xc96f2715e2a242d50d1b0bc923dbe1740b8ecf18`]
+which is not the canonical USDC on Arbitrum. The canonical USDC on Arbitrum is
+link:https://arbiscan.io/token/0xff970a61a04b1ca14834a43f5de4533ebddb5cc8[`0xff970a61a04b1ca14834a43f5de4533ebddb5cc8`].
+
+Wormhole cannot magically mint all tokens on Arbitrum or other L2s. Wormhole
+mints its own token representing the bridged asset that may be swapped to the
+canonical representation on the given L2 assuming there is a market for such
+a swap. The overview of liquid markets on each target chain is available in the
+link:https://docs.wormhole.com/wormhole/overview-liquid-markets[Wormhole documentation].
+Not every chain and token pair has a market. 
+
+To fully use the capabilities of the Wormhole portal and make the user
+experience seamless, we will implement a gateway contract having the authority to
+wrap Wormhole-specific tokens into tBTC tokens in a 1:1 ratio. This contract
+will have a minting authority for tBTC on L2. This way, no liquid market has to
+exist on any target chain for users to be able to cross the Wormhole bridge into
+the canonical tBTC.
+
+IMPORTANT: The requirement for a seamless user experience is that the Wormhole
+team has to integrate the step of wrapping the token using the gateway contract
+into the Wormhole bridge flow, in the same transaction as the L2 confirmation.
+
+There are two ways to implement a gateway: use plain token transfers or custom
+messages for mint and burn.
+
+===== Plain token transfer gateway
+
+This is the easiest approach and the gateway contract can be integrated directly
+with the token bridge. This approach will benefit from Wormhole security
+measures like various accounting invariant checks and governable notional limits
+between chains. Also, the integration into the UI will be easier and cross-chain
+transfers will be automatically relayed. Last but not least, any possible
+security issues with Wormhole bridge are limited to the amount of tBTC bridged
+from L1 to L2 so far.
```

Comment:
They are limited to a single L2 chain or to all Wormhole-powered L2 chains tBTC can be minted on? In other words, what's the exact impact of a security issue?

Reply:
This section is gone now - see https://github.com/keep-network/tbtc-v2/pull/542/commits/fc9328355c65ddfb36d2820b4f895e085a4b6323 - and we may discuss it more during the review of the contract implementing the gateway. But let me also explain here.

With the approach as presented here, where every separate bridge has its own gateway on L1, the risk of a hack of the given gateway is limited to the amount of tokens that have been bridged from L1 to L2 through the given gateway.